### PR TITLE
Fully working rebindable keys & reorganising of some files

### LINKED ISSUE #953: Redesign Input system B: 70
# Preliminary information

## Category
TODO

# Report
## Current Behavior
Input stuff is currently hardcoded and scattered throughout the codebase

## Expected/Wanted/Requested Behavior

- Avoid hardcoded constructions like this one:
```
			if (azerty)
			{
				if (Input.GetKeyDown(KeyCode.A))
				{
					Drop();
				}
			} else {
				if (Input.GetKeyDown(KeyCode.Q))
				{
					Drop();
				}
			}
```
- Centralize all input controls (key/mouse/touch) in some sort of input manager?
- Allow custom keybindings

Suggestions are welcome.

### PR DESCRIPTION
**Pull request by Jibbajabbafic**
### Purpose
Fixes #953

Sorry for dumping so many changes into one branch. I really should have separated them.

- Movement keys can now be rebound like other keys.
- Numpad8 now cycles between head, eyes and mouth
- Adds eyes and mouth to targeting (doesn't do anything yet)
- Renamed intents to be more consistent with original SS13
- Realigns some of the zone selector boxes and action buttons
- Moves Scripts/PlayGroup/Input to Scripts/Input System along with all keybind files
- Renamed PlayGroup folder to Player
- Renames InputController to MouseInputController
- Removes AZERTY stuff and settings ClientTab

### Open Questions and Pre-Merge TODOs

- [x]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [ ]  This PR has been tested in multiplayer

### CODE REVIEW

**Comment by Jibbajabbafic** [comment]
Oh yeah also, weapons now reload with Z and throwing is back to being toggled (undoing my very first contribution lol)

**Review by fomalsd — COMMENTED** [review]
Damn, that's outstanding! Such a relief to have a proper input system in place :)
(yet to test it in game)

**Review by DooblyNoobly — COMMENTED** [review]
image

IngameMenu has a Network Identity attached to it but it is also apart of the Managers prefab. NetID's cannot be on child objects and also on scene objects that travel between scenes (DoNotDestroy). This caused a bug where clients could not open the In Game Menu because unet disabled the object. I'll merge this PR and quickly follow it with a fix to this problem to quickly get the managers.prefab changes through

File: UnityProject/Assets/Scripts/Input System/KeyboardInputManager.cs [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+	}
+
+	void Update()
+	{
+		CheckKeyboardInput();
+	}
+
+	void CheckKeyboardInput()
+	{
+		if (!UIManager.IsInputFocus && GameData.IsInGame && CustomNetworkManager.Instance.IsClientConnected())
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/Input System/KeyboardInputManager.cs:36]: A note: `CustomNetworkManager.Instance.IsClientConnected()` is now obsolete. But that is not a problem here as we are waiting to see what the new unet holds before converting everything

File: UnityProject/Assets/Scripts/UI/IngameMenu/GUI_IngameMenu.cs [hunk 2]
```diff
@@ -29,30 +29,6 @@ void Awake()
 		}
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/UI/IngameMenu/GUI_IngameMenu.cs:29]: This class derives from NetworkBehaviour but it is on a child object of managers.prefab. I'll be converting this back to a monobehaviour with a follow up fix

**Pull request merged**